py/vtdb/vtgatev2.py

```diff
@@ -65,18 +38,32 @@ def handle_app_error(exc_args):
   return dbexceptions.DatabaseError(exc_args)
 
 
-@log_exception
-def convert_exception(exc, *args):
+def convert_exception(exc, *args, **kwargs):
   new_args = exc.args + args
+  if kwargs:
+    new_args += tuple(kwargs.values())
+  new_exc = exc
+
   if isinstance(exc, gorpc.TimeoutError):
-    return dbexceptions.TimeoutError(new_args)
+    new_exc = dbexceptions.TimeoutError(new_args)
   elif isinstance(exc, gorpc.AppError):
-    return handle_app_error(new_args)
+    new_exc = handle_app_error(new_args)
   elif isinstance(exc, gorpc.ProgrammingError):
-    return dbexceptions.ProgrammingError(new_args)
+    new_exc = dbexceptions.ProgrammingError(new_args)
   elif isinstance(exc, gorpc.GoRpcError):
-    return dbexceptions.FatalError(new_args)
-  return exc
+    new_exc = dbexceptions.FatalError(new_args)
+
+  keyspace = None
+  tablet_type = None
+  try:
+    keyspace = kwargs["keyspace"]
+    tablet_type = kwargs["tablet_type"]
+  except KeyError:
+    pass
```

Comment:
Why not replace the whole block above this with:

keyspace = kwargs.get("keyspace", None)
tablet_type = kwargs.get("tablet_type", None)